src/renderer/wg_engine/tvgWgPipelines.cpp

```diff
@@ -223,33 +223,171 @@ void WgPipelineImage::initialize(WGPUDevice device)
+    WGPUCompareFunction stencilFuncion = WGPUCompareFunction_Always;
+    WGPUStencilOperation stencilOperation = WGPUStencilOperation_Zero;
+
+    // sheder source and labels
+    auto shaderSource = cShaderSource_PipelineBlitColor;
+    auto shaderLabel = "The shader blit color";
+    auto pipelineLabel = "The render pipeline blit color";
+
+    // allocate all pipeline handles
+    allocate(device,
+             vertexBufferLayouts, ARRAY_ELEMENTS_COUNT(vertexBufferLayouts),
+             bindGroupLayouts, ARRAY_ELEMENTS_COUNT(bindGroupLayouts),
+             stencilFuncion, stencilOperation,
+             shaderSource, shaderLabel, pipelineLabel);
+}
+
+
+void WgPipelineComposition::initialize(WGPUDevice device, const char* shaderSrc)
+{
+    // vertex and buffers settings
+    WGPUVertexAttribute vertexAttributesPos = { WGPUVertexFormat_Float32x2, sizeof(float) * 0, 0 };
+    WGPUVertexAttribute vertexAttributesTex = { WGPUVertexFormat_Float32x2, sizeof(float) * 0, 1 };
+    WGPUVertexBufferLayout vertexBufferLayouts[] = {
+        makeVertexBufferLayout(&vertexAttributesPos, 1, sizeof(float) * 2),
+        makeVertexBufferLayout(&vertexAttributesTex, 1, sizeof(float) * 2)
+    };
+
+    // bind groups and layouts
+    WGPUBindGroupLayout bindGroupLayouts[] = {
+        WgBindGroupBlit::getLayout(device),
+        WgBindGroupBlit::getLayout(device)
+    };
+
+    // stencil function
+    WGPUCompareFunction stencilFuncion = WGPUCompareFunction_Always;
+    WGPUStencilOperation stencilOperation = WGPUStencilOperation_Zero;
+
+    // sheder source and labels
+    auto shaderSource = shaderSrc;
+    auto shaderLabel = "The shader compose alpha mask";
+    auto pipelineLabel = "The render pipeline compose alpha mask";
+
+    // allocate all pipeline handles
+    allocate(device,
+             vertexBufferLayouts, ARRAY_ELEMENTS_COUNT(vertexBufferLayouts),
+             bindGroupLayouts, ARRAY_ELEMENTS_COUNT(bindGroupLayouts),
+             stencilFuncion, stencilOperation,
+             shaderSource, shaderLabel, pipelineLabel);
+}
+
 //************************************************************************
 // pipelines
 //************************************************************************
 
 void WgPipelines::initialize(WGPUDevice device)
 {
-    mPipelineFillShape.initialize(device);
-    mPipelineFillStroke.initialize(device);
-    mPipelineSolid.initialize(device);
-    mPipelineLinear.initialize(device);
-    mPipelineRadial.initialize(device);
-    mPipelineImage.initialize(device);
+    fillShape.initialize(device);
+    fillStroke.initialize(device);
+    solid.initialize(device);
+    linear.initialize(device);
+    radial.initialize(device);
+    image.initialize(device);
+    blit.initialize(device);
+    blitColor.initialize(device);
+    // composition pipelines
+    compAlphaMask.initialize(device, cShaderSource_PipelineCompAlphaMask);
+    compInvAlphaMask.initialize(device, cShaderSource_PipelineCompInvAlphaMask);
```

Comment:
Except for alpha masking, the other features are quite optional. They are not frequently used in many animations. That would be better if we can initialize them in a deffered way.

Reply:
> Except for alpha masking, the other features are quite optional. They are not frequently used in many animations. That would be better if we can initialize them in a deffered way.

Good point. Will use RAII for it